File: programmers/Lv.2/올바른 괄호.py
def solution(s):
    
    stack = []
    
    for ch in s :
        if len(stack) != 0 and stack[-1] == "(" and ch == ")" :
            stack.pop()
        else :
            stack.append(ch)
    
    if len(stack) != 0 :
        return False
    else :
        return True

File: Programmers/Javascript/Lv.2/올바른 괄호.js
function solution(s) {
  const stack = [];

  [...s].forEach((el, idx) => {
    if (idx === 0) stack.push(el);
    else {
      if (stack[stack.length - 1] === "(" && el === ")") {
        stack.pop();
      } else stack.push(el);
    }
  });

  return stack.length === 0 ? true : false;
}
